src/main/java/vct/col/veymont/TerminationCheck.scala

```diff
@@ -0,0 +1,199 @@
+      }
+    }
+  }
+
+  def MethodInvToString(mi : MethodInvokation) = mi.method //(if (mi.`object`==null) "" else mi.`object`.toString) + mi.method
+
+  override def visit(l : LoopStatement) : Unit =
+    if(currentClass != mainClassName)
+      Warning("Session Warning: loop in method of non-Main class" + deadlockWarning)
+
+  override def visit(s : ASTSpecial) : Unit = {
+    s.kind match {
+      case ASTSpecial.Kind.Goto => Warning("Session Warning: Goto" + deadlockWarning)
+      case ASTSpecial.Kind.Wait => Warning("Session Warning: Wait" + deadlockWarning)
+      case ASTSpecial.Kind.Fork => Warning("Session Warning: Fork" + deadlockWarning)
+      case ASTSpecial.Kind.Join => Warning("Session Warning: Join" + deadlockWarning)
+      case ASTSpecial.Kind.Lock => Warning("Session Warning: Lock" + deadlockWarning)
+      case ASTSpecial.Kind.Unlock => Warning("Session Warning: Unlock" + deadlockWarning)
+      case ASTSpecial.Kind.Send => Warning("Session Warning: Send" + deadlockWarning)
+      case ASTSpecial.Kind.Recv => Warning("Session Warning: Recv" + deadlockWarning)
+      case _ => super.visit(s)
+    }
+  }
+/*
+  private def isForLoop(l : LoopStatement) : Boolean = {
+    val guardOp = l.getEntryGuard.asInstanceOf[OperatorExpression]
+    val initval = l.getInitBlock.asInstanceOf[BlockStatement].getStatement(0).asInstanceOf[VariableDeclaration].get().head.asInstanceOf[DeclarationStatement].init.get.asInstanceOf[ConstantExpression]
+
+    val updateOp = l.getUpdateBlock.asInstanceOf[BlockStatement].getStatement(0)
+    //TODO: get Value from initval
+    if(isLengthExpr(guardOp.arg(1)))
+      initval.value == 0 && Set(StandardOperator.LT).contains(guardOp.operator) && updateOp.operator == StandardOperator.PostIncr
+    else initval.value < guardOp.arg(1).asInstanceOf[ConstantExpression].value && updateOp.operator == StandardOperator.PostIncr ||
+      initval.value > guardOp.arg(1).asInstanceOf[ConstantExpression].value && updateOp.operator == StandardOperator.PostDecr
+  }
+*/
+  private def isLengthExpr(e : ASTNode) : Boolean = e match {
+    case d : Dereference => d.field == Dereference.ArrayLength
+    case _ => false
+  }
+
+  //TODO: check declaration has int type
+  private def isConstantInitDecl(i : ASTNode) : Boolean =
+    i match {
+      case ib : BlockStatement =>
+        ib.size == 1 && (ib.getStatement(0) match {
+          case init : VariableDeclaration =>
+            init.basetype match {
+              case p : PrimitiveType => p.isInteger && init.get().size == 1 && (init.get().head match {
+                case d : DeclarationStatement => d.init match {
+                  case Some(initval) => initval.isInstanceOf[ConstantExpression]
+                  case None => false
+                }
+                case _ => false
+              })
+              case _ => false
+            }
+          case _ => false
+        })
+      case _ => false
```

Comment:
I never know what to do about these silly `case _ => default` chains...

Reply:
I removed this code because it was not used.